includes/wikia/api/DesignSystemApiController.class.php

```diff
@@ -1,21 +1,185 @@
 <?php
 
 class DesignSystemApiController extends WikiaApiController {
+	private $wikiId;
+	private $lang;
+
+	private $data = [
+		'brand-logo' => [
+			'type' => 'brand-logo',
+			'href' => '#',
+			'elements' => [
+				[
+					'type' => 'line-image',
+					'image' => 'company/logo-fandom',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-logo-fandom',
+					]
+				],
+				[
+					'type' => 'line-image',
+					'image' => 'company/logo-wikia',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-logo-wikia',
+					]
+				],
+				[
+					'type' => 'translatable-text',
+					'key' => 'global-navigation-header-title',
+				],
+			],
+		],
+		'brand-links' => [
+			[
+				'type' => 'link-branded',
+				'brand' => 'tv',
+				'title' => [
+					'type' => 'translatable-text',
+					'key' => 'global-navigation-brandlink-vertical-tv'
+				],
+				'href' => 'http:\/\/tv.wikia.com'
+			],
+			[
+				'type' => 'link-branded',
+				'brand' => 'games',
+				'title' => [
+					'type' => 'translatable-text',
+					'key' => 'global-navigation-brandlink-vertical-games'
+				],
+				'href' => 'http:\/\/games.wikia.com'
+			],
+			[
+				'type' => 'link-branded',
+				'brand' => 'movies',
+				'title' => [
+					'type' => 'translatable-text',
+					'key' => 'global-navigation-brandlink-vertical-movies'
+				],
+				'href' => 'http:\/\/movies.wikia.com'
+			],
+			[
+				'type' => 'links-branded',
+				'brand' => 'wikis',
+				'title' => [
+					'type' => 'translatable-text',
+					'key' => 'global-navigation-header-title',
+				],
+				'links' => [ // not branded
+					[
+						'type' => 'link-branded',
+						'brand' => 'wikis',
+						'title' => [
+							'type' => 'translatable-text',
+							'key' => 'global-navigation-brandlink-vertical-wikis'
+						],
+						'href' => '#'
+					],
+					[
+						'type' => 'link-branded',
+						'brand' => 'communitycentral',
+						'title' => [
+							'type' => 'translatable-text',
+							'key' => 'global-navigation-brandlink-vertical-communitycentral'
+						],
+						'href' => '#'
+					]
+				]
+			]
+		],
+		'search' => [
+			[
+				'type' => 'search-endpoint',  // new_type
+				'endpoint-href' => 'some.search.link',
+				'suggestions-href' => 'suggestions.link', // make object
+				'placeholder-inactive' => [
+					'type' => 'translatable-text',
+					'key' => 'global-navigation-search-placeholder-inactive'
+				],
+				'placeholder-active' => [
+					'type' => 'translatable-text',
+					'key' => 'global-navigation-search-placeholder-active'
+				],
+				'title' => [
+					'type' => 'translatable-text',
+					'key' => 'global-navigation-search-title'
+				]
+			]
+		],
+		'user-info' => [
+			'type' => 'user-not-authenticated', // new_type
+			'avatar' => [ ],
+			'links' => [
+				[
+					'type' => 'link-text',
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "link-text" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)